wpilibc/src/main/native/cpp/AnalogGyro.cpp

```diff
@@ -64,6 +65,18 @@ AnalogGyro::AnalogGyro(std::shared_ptr<AnalogInput> channel, int center,
 
 AnalogGyro::~AnalogGyro() { HAL_FreeAnalogGyro(m_gyroHandle); }
 
+AnalogGyro::AnalogGyro(AnalogGyro&& rhs) {
+  m_analog = std::move(rhs.m_analog);
+  std::swap(m_gyroHandle, rhs.m_gyroHandle);
+}
+
+AnalogGyro& AnalogGyro::operator=(AnalogGyro&& rhs) {
+  m_analog = std::move(rhs.m_analog);
+  std::swap(m_gyroHandle, rhs.m_gyroHandle);
+
+  return *this;
+}
```

Comment:
For the more complex classes, rather than duplicating, would it be better to define a swap() function and then use it in both the move constructor and operator=?  However, that approach does require a no-args constructor be available, and I'm not sure if we want that, although it can be useful in any case.  See cscore_oo.h/inl, e.g. for VideoSource.

Reply:
No-arg constructor is pretty nice to have for move semantics in general (like move-only socket classes or std::thread). Hm...